### DISCUSSION TITLE
KlipperScreen gradually slows to a crawl/becomes unresponsive after some time, is this a known issue?

### DISCUSSION THREAD

**Discussion by dburr** [KlipperScreen/KlipperScreen#1391] [Q&A]
Pi 4 2 GB, generic 5" touchscreen, minimal install of Raspbian 11 Bullseye (just Klipper + Moonraker + Mainsail + KlipperScreen + Mobileraker). After a few days of uptime, the screen will start to become more and more sluggish to respond to taps or to scroll, and will eventually become completely unresponsive. If I restart the KlipperScreen service, everything comes back and works perfectly for another few days, then the whole cycle repeats itself. This doesn't affect the Pi itself, or Klipper; printing via web still works perfectly. It's just the KlipperScreen screen that becomes unresponsive. For now I have set up a cron job to restart the KlipperScreen service daily, which works around the issue. Is this a known issue, and is there any fix? Feels like some sort of gradual memory leak to me. As far as I can tell I'm not doing anything particularly strange or exotic in my KlipperScreen config. The only thing I added was an indicator to show the Pi's temperature (which I have configured Klipper to monitor.)

**Comment by alfrix** [comment]
i will investigate, do you use multiple printers and switch between them?

**Reply by winningfaith81** [reply]
To revive this thread, I have a similar issue. If I turn my printer on in the morning when I get up, KlipperScreen won't be responsive by lunch. This has been happening for a while. Stopping and restarting KlipperScreen within Mainsail doesn't see to fix it. I have to reboot my Raspberry Pi. I only have 1 printer. 

I am on: 
Klipper v0.12.0-418-g0114d72a
Mainsail v2.13.2
KlipperScreen v0.4.5-40

Raspberry Pi 5
BTT PI TFT70 v2.1
